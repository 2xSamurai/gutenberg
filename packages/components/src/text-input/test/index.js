/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { TextInput } from '../index';

describe( 'props', () => {
	let base;

	beforeEach( () => {
		( { container: base } = render( <TextInput /> ) );
	} );
	test( 'should render correctly', () => {
		expect( base.firstChild ).toMatchSnapshot();
	} );

	test( 'should render disabled', () => {
		const { container } = render( <TextInput disabled /> );
		expect( container.firstChild ).toMatchDiffSnapshot( base.firstChild );
	} );

	test( 'should render defaultValue', () => {
		const { container } = render(
			<TextInput defaultValue="WordPress.org" />
		);
		expect( container.firstChild ).toMatchDiffSnapshot( base.firstChild );
	} );

	// TODO: TextareaAutosizeClass is creating invalid attributes.
	test( 'should render multiline', () => {
		const { container } = render( <TextInput isResizable multiline /> );
		expect( container.firstChild ).toMatchDiffSnapshot( base.firstChild );
	} );

	test( 'should render prefix', () => {
		const { container } = render(
			<TextInput prefix={ <span>prefix</span> } />
		);
		expect( container.firstChild ).toMatchDiffSnapshot( base.firstChild );
	} );

	test( 'should render size', () => {
		const { container } = render( <TextInput size="small" /> );
		expect( container.firstChild ).toMatchDiffSnapshot( base.firstChild );
	} );

	test( 'should render suffix', () => {
		const { container } = render(
			<TextInput suffix={ <span>suffix</span> } />
		);
		expect( container.firstChild ).toMatchDiffSnapshot( base.firstChild );
	} );

	test( 'should render value', () => {
		const { container } = render( <TextInput value="WordPress.org" /> );
		expect( container.firstChild ).toMatchDiffSnapshot( base.firstChild );
	} );
} );
