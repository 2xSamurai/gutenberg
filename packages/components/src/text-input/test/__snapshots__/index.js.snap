// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props should render correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  background: #fff;
  border-radius: 2px;
  border: 1px solid;
  border-color: #757575;
  box-shadow: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 13px;
  outline: none;
  padding: 0 8px;
  position: relative;
  -webkit-transition: border-color 100ms ease;
  transition: border-color 100ms ease;
  width: 100%;
}

.emotion-0>*+*:not(marquee) {
  margin-left: calc(4px * 2);
}

.emotion-0>* {
  min-width: 0;
}

@media ( prefers-reduced-motion: reduce ) {
  .emotion-0 {
    transition-duration: 0ms;
  }
}

.emotion-0[disabled] {
  opacity: 0.6;
}

.emotion-0:hover {
  border-color: #757575;
}

.emotion-0:focus,
.emotion-0[data-focused='true'] {
  border-color: var( --wp-admin-theme-color, #00669b);
  box-shadow: 0 0 0,0.5px,[object Object];
}

.emotion-2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: transparent;
  border: none;
  border-radius: 2px;
  box-shadow: none;
  box-sizing: border-box;
  color:100: #8f98a1;
  150: #8d96a0;
  200: #7e8993;
  300: #6c7781;
  400: #606a73;
  500: #555d66;
  600: #40464d;
  700: #32373c;
  800: #23282d;
  900: #191e23;
  placeholder: rgba(30, 30, 30, 0.62);
  primary: #1e1e1e;
  heading: #050505;
  display: block;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  line-height: 18px;
  min-height: calc( 36px - 2px );
  outline: none;
  padding: 0;
  padding-bottom: calc( ( 36px - 2px - 18px ) / 2 );
  padding-top: calc( ( 36px - 2px - 18px ) / 2 );
  resize: none;
  text-align: left;
  width: 100%;
  font-size: 16px;
  min-height: calc( calc( 36px * 0.1 ) - 2px );
  padding-bottom: calc(
  		( calc( 36px * 0.1 ) - 2px - 18px ) / 2
  	);
  padding-top: calc( ( calc( 36px * 0.1 ) - 2px - 18px ) / 2 );
}

@media ( min-width: 36em ) {
  .emotion-2 {
    font-size: 13px;
  }
}

.emotion-2::-webkit-outer-spin-button,
.emotion-2::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.emotion-2[type='number'] {
  -moz-appearance: textfield;
}

<div
  class="components-flex components-base-field components-text-input emotion-0 emotion-1"
  data-wp-c16t="true"
  data-wp-component="BaseField"
>
  <input
    autocomplete="off"
    class="emotion-2 emotion-1"
    data-wp-c16t="true"
    data-wp-component="TextInput"
    spellcheck="false"
    step="1"
    type="text"
    value=""
  />
</div>
`;

exports[`props should render defaultValue 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -9,8 +9,8 @@
      data-wp-c16t="true"
      data-wp-component="TextInput"
      spellcheck="false"
      step="1"
      type="text"
-     value="WordPress.org"
+     value=""
    />
  </div>
`;

exports[`props should render disabled 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,17 +1,15 @@
  <div
    class="components-flex components-base-field components-text-input css-137jbvx-View-Flex-sx-Base-sx-Items-ItemsRow-BaseField em57xhy0"
    data-wp-c16t="true"
    data-wp-component="BaseField"
-   disabled=""
  >
    <input
      autocomplete="off"
      class="css-he2ew4-View-inputFontSize-Input-medium em57xhy0"
      data-wp-c16t="true"
      data-wp-component="TextInput"
-     disabled=""
      spellcheck="false"
      step="1"
      type="text"
      value=""
    />
`;

exports[`props should render multiline 1`] = `
Snapshot Diff:
- First value
+ Second value

  <div
-   class="components-flex components-base-field components-text-input css-1bavlj-View-Flex-sx-Base-sx-Items-ItemsRow-BaseField-multiline em57xhy0"
+   class="components-flex components-base-field components-text-input css-137jbvx-View-Flex-sx-Base-sx-Items-ItemsRow-BaseField em57xhy0"
    data-wp-c16t="true"
    data-wp-component="BaseField"
  >
-   <textarea
+   <input
      autocomplete="off"
-     class="css-1weg09x-View-inputFontSize-Input-medium-inputMultiline-resizable-scrollableScrollbar em57xhy0"
+     class="css-he2ew4-View-inputFontSize-Input-medium em57xhy0"
      data-wp-c16t="true"
      data-wp-component="TextInput"
-     rows="1"
      spellcheck="false"
      step="1"
-     style="overflow-x: hidden; word-wrap: break-word; resize: none; overflow-y: hidden;"
      type="text"
+     value=""
    />
  </div>
`;

exports[`props should render prefix 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,13 +1,10 @@
  <div
    class="components-flex components-base-field components-text-input css-137jbvx-View-Flex-sx-Base-sx-Items-ItemsRow-BaseField em57xhy0"
    data-wp-c16t="true"
    data-wp-component="BaseField"
  >
-   <span>
-     prefix
-   </span>
    <input
      autocomplete="off"
      class="css-he2ew4-View-inputFontSize-Input-medium em57xhy0"
      data-wp-c16t="true"
      data-wp-component="TextInput"
`;

exports[`props should render size 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -3,11 +3,11 @@
    data-wp-c16t="true"
    data-wp-component="BaseField"
  >
    <input
      autocomplete="off"
-     class="css-1sazt8m-View-inputFontSize-Input-small em57xhy0"
+     class="css-he2ew4-View-inputFontSize-Input-medium em57xhy0"
      data-wp-c16t="true"
      data-wp-component="TextInput"
      spellcheck="false"
      step="1"
      type="text"
`;

exports[`props should render suffix 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -11,9 +11,6 @@
      spellcheck="false"
      step="1"
      type="text"
      value=""
    />
-   <span>
-     suffix
-   </span>
  </div>
`;

exports[`props should render value 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -9,8 +9,8 @@
      data-wp-c16t="true"
      data-wp-component="TextInput"
      spellcheck="false"
      step="1"
      type="text"
-     value="WordPress.org"
+     value=""
    />
  </div>
`;
